azurelinuxagent/ga/exthandlers.py

```diff
@@ -296,6 +296,11 @@ def _incarnation_changed(self, etag):
         # Skip processing if GoalState incarnation did not change
         return self.last_etag != etag
 
+    def __last_gs_unsupported(self):
```

Comment:
The code will work as is, but the return value of this function is valid only after the aggregate status has been initialized. Consider raising an exception if it is called before that.

Also, the fact that there are unsupported features is more a property of the goal state itself, not of the aggregate status or the ExtHandlersHandler. We can leave as is an do that on V3, if you prefer so.

Reply:
>but the return value of this function is valid only after the aggregate status has been initialized.

This would always be initialized, we're initializing a default value in `__init__`

```python
self.__gs_aggregate_status = GoalStateAggregateStatus(status=GoalStateStatus.Initialize, seq_no="-1",
                                                              code=GoalStateAggregateStatusCodes.Success,
                                                              message="Initializing new GoalState")
```

>Also, the fact that there are unsupported features is more a property of the goal state itself, not of the aggregate status or the ExtHandlersHandler.

I totally agree with this, I was contemplating making it a property of the GS but decided against it as it would need extra code changes which would not add a lot of value. I agree with making it part of V3, we can just design it accordingly.

Reply:
It is initialized, but the value doesn't have an actual valid meaning, since the computation for unsupported features is done later. At this point the best one can say is 'unknown'
Ok to do in V3